src/module/SafeModeValuePersistence.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2021 Wolkabout s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "module/SafeModeValuePersistence.h"
+
+#include "core/utilities/FileSystemUtils.h"
+#include "core/utilities/Logger.h"
+#include "core/utilities/json.hpp"
+
+using namespace nlohmann;
+
+namespace wolkabout
+{
+const std::string JSON_PATH = "./safe-mode.json";
+const std::string SEPARATOR = ".";
+
+bool SafeModeValuePersistence::storeSafeModeValue(const std::string& deviceKey, const std::string& reference,
+                                                  std::string value)
+{
+    LOG(TRACE) << METHOD_INFO;
+
+    // Check if the file exists
+    auto j = json{};
+    if (FileSystemUtils::isFilePresent(JSON_PATH))
+    {
+        // Read the content of the file
+        auto content = std::string{};
+        if (FileSystemUtils::readFileContent(JSON_PATH, content))
+        {
+            try
+            {
+                j = json::parse(content);
+            }
+            catch (...)
+            {
+            }
+        }
+    }
+
+    // Now change the value in the json
+    j[deviceKey + SEPARATOR + reference] = value;
+
+    // And now try to write it in
+    return FileSystemUtils::createFileWithContent(JSON_PATH, j.dump(4));
```

Comment:
Will this rewrite file contents if exists?

Reply:
Yes, but all the content was also loaded. You kind off have to read the entire file, edit it, and then rewrite it.